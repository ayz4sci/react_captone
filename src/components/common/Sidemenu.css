/* Position and sizing of burger button */
.bm-burger-button {
  position: fixed;
  width: 28px;
  height: 25px;
  left: 36px;
  top: 36px;
}
 
/* Color/shape of burger icon bars */
.bm-burger-bars {
  background: #373a47;
}
 
/* Position and sizing of clickable cross button */
.bm-cross-button {
  height: 24px;
  width: 24px;
}
 
/* Color/shape of close button cross */
.bm-cross {
  background: #222222;
  width: 80%;
}
 
/* General sidebar styles */
.bm-menu {
  background: #fcfcfc;
  padding: 2.5em 1.5em 0;
  font-size: 1.15em;
}