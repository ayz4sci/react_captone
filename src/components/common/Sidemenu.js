import React from 'react';
import PropTypes from 'prop-types';
import BurgerMenu from 'react-burger-menu';
import LoginForm from '../authentication/Login';
import './Sidemenu.css';
import UserProfilePage from '../authentication/UserProfilePage';

class Sidemenu extends React.Component{
    Menu = BurgerMenu['slide'];

    state = {
        errors: {},
        username: "",
        password: "",
        loading: this.props.loading,
        user: this.props.user
    }

    componentWillReceiveProps = (nextProps) => {
        if (nextProps.user !== this.state.user) {
            this.setState({ 
                loading: false,
                user: nextProps.user
            });

        }
    }

    onChange = (e) => {
        this.setState({ [e.target.name] : e.target.value});
    }

    onEnter = (e) => {
        e.preventDefault();
        if(!this.formIsValid()) return;
        this.setState({loading: true});
        this.props.userActions.loadUser(this.state.username, this.state.password);
    }

    onSignOut = (e) => {
        e.preventDefault();
        this.setState({user: null});
    }

    formIsValid = () => {
        let formIsValid = true;
        let errors = {};

        if(this.state.username.length < 5){
            errors.username = 'Username must be at least 5 characters.';
            formIsValid = false;
        } else if(this.state.password.length < 5){
            errors.password = 'Password must be at least 5 characters.';
            formIsValid = false;
        }

        this.setState({errors: errors});
        return formIsValid;
    }

    render(){
        return (
            <this.Menu 
                pageWrapId={ "page-wrap" } 
                outerContainerId={ "outer-container" } 
                slide="true">
                { !this.state.user ? 
                    <LoginForm 
                        {...this.state}
                        onChange={this.onChange}
                        onEnter={this.onEnter}/>
                    :
                    <UserProfilePage 
                        username={this.state.username}
                        onSignOut={this.onSignOut} />
                }
            </this.Menu>
        )
    }
}

Sidemenu.propTypes = {
    userActions: PropTypes.object,
    user: PropTypes.object
}

export default Sidemenu;