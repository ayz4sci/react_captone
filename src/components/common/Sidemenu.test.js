import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Sidemenu from './Sidemenu';

it('renders correctly', () => {
    const props = { 
        userActions: {},
        user: {}
    };

    const wrapper = shallow(<Sidemenu {...props} />, {mode: 'shallow'})
    expect(toJson(wrapper)).toMatchSnapshot();
});