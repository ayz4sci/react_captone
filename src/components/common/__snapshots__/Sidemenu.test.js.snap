// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Menu
  bodyClassName=""
  burgerBarClassName=""
  burgerButtonClassName=""
  className=""
  crossButtonClassName=""
  crossClassName=""
  disableCloseOnEsc={false}
  id=""
  itemClassName=""
  itemListClassName=""
  menuClassName=""
  morphShapeClassName=""
  noOverlay={false}
  onStateChange={[Function]}
  outerContainerId="outer-container"
  overlayClassName=""
  pageWrapId="page-wrap"
  slide="true"
  styles={Object {}}
  width={300}
>
  <UserProfilePage
    onSignOut={[Function]}
    username=""
  />
</Menu>
`;
